File: lib/models/series.dart
class Series {
  final String id;
  final String name;
  final String? description;
  final String? profilepic;
  final String? banner;
  final String? genre;
  final String? language;
  final List<String> tags;
  final int episodeCount;
  final DateTime createdAt;
  final DateTime updatedAt;

  Series({
    required this.id,
    required this.name,
    this.description,
    this.profilepic,
    this.banner,
    this.genre,
    this.language,
    required this.tags,
    required this.episodeCount,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Series.fromJson(Map<String, dynamic> json) {
    return Series(
      id: json['_id'],
      name: json['name'],
      description: json['description'] ?? json['seriesDescription'],
      profilepic: json['profilepic'],
      banner: json['banner'],
      genre: json['genre'],
      language: json['language'],
      tags: List<String>.from(json['tags'] ?? []),
      episodeCount: json['episodeCount'] ?? json['totalEpisodes'] ?? 0,
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}

File: backend/models/Series.js
const mongoose = require('mongoose');

console.log('📝 Loading Series model...');

const seriesSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true
  },
  description: {
    type: String,
    default: ''
  },
  profilepic: {
    type: String,
    default: ''
  },
  banner: {
    type: String,
    default: ''
  },
  genre: {
    type: String,
    default: 'General'
  },
  language: {
    type: String,
    default: 'Hindi'
  },
  tags: [{
    type: String
  }],
  isActive: {
    type: Boolean,
    default: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

seriesSchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

const Series = mongoose.model('Series', seriesSchema);

console.log('✅ Series model loaded successfully');

module.exports = Series;
